### PR TITLE
Imagestream naming

### PR DESCRIPTION
**Pull request by bkabrda**
This PR makes sure we get three things right:
- `imagestream` argument for `ImportImagePlugin`
- `docker_image_repo` argument for `ImportImagePlugin`
- name of trigger that built image depends on

Also, the second commit fixes minor issue, where openshift url for `check_and_set_rebuild` wasn't previously substituted in `BuildRequest`.

### CODE REVIEW

**Comment by bkabrda** [comment]
The thing I'm not sure about is whether we want to limit osbs-client to Dockerfiles that have `Name` label specified (which this PR effectively does). IMO this is ok, but if someone thinks otherwise, speak up.

**Comment by twaugh** [comment]
I've made a separate pull request for the `check_and_set_rebuild` fix, so that this pull request can be examined independently.

**Comment by bkabrda** [comment]
I'll remote the commit from this PR, then. (Just for reference, the commit is now in PR #195).

**Comment by twaugh** [comment]
Just a note: the BuildConfigs and ImageStreams are both tracking the same units, so they ought to use the same names as each other.

**Comment by bkabrda** [comment]
Done, should be ready for merge.

File: osbs/build/build_request.py [hunk 1]
```diff
@@ -185,6 +185,14 @@ def render(self):
                 self.dj.dock_json_set_arg('postbuild_plugins', "store_metadata_in_osv3",
                                           "use_auth", self.spec.use_auth.value)
 
+        if self.dj.dock_json_has_plugin_conf('postbuild_plugins', 'import_image'):
+            self.dj.dock_json_set_arg('postbuild_plugins', 'import_image', 'imagestream',
+                                      self.spec.imagestream_name.value)
+            self.dj.dock_json_set_arg('postbuild_plugins', 'import_image', 'docker_image_repo',
+                                      self.spec.imagestream_url.value)
+            self.dj.dock_json_set_arg('postbuild_plugins', 'import_image', 'url',
+                                      self.spec.openshift_uri.value)
```
Comment (twaugh) [inline osbs/build/build_request.py]: I think we also want to set `use_auth` here, don't we? cf. line 182 further up.

File: inputs/simple_inner.json [hunk 2]
```diff
@@ -29,6 +29,15 @@
...
       "name": "all_rpm_packages"
+    },
+    {
+      "args": {
+        "imagestream": "{{IMAGESTREAM}}",
+        "docker_image_repo": "{{DOCKER_IMAGE_REPO}}",
+        "url": "{{OPENSHIFT_URI}}",
+        "verify_ssl": false
+      },
+      "name": "import_image"
```
Comment (twaugh) [inline inputs/simple_inner.json]: Does the simple build type get all the rebuild trigger bits? Maybe that should only be in the prod type.

File: osbs/build/spec.py [hunk 3]
```diff
@@ -144,8 +147,11 @@ def set_params(self, git_uri=None, git_ref=None, git_branch=None, registry_uri=N
             raise OsbsValidationException("yum_repourls must be a list")
         self.yum_repourls.value = yum_repourls or []
         self.use_auth.value = use_auth
-        self.name.value = '%s-%s' % (self.component.value, self.git_branch.value)
-        self.base_image.value = os.path.join(registry_uri, base_image)
+        self.name.value = name_label.replace('/', '-')
+        self.trigger_imagestream_name.value = get_imagestream_name_from_image(base_image)
+        # imagestream and buildconfig have precisely the same name
+        self.imagestream_name.value = self.name.value
```
Comment (twaugh) [inline osbs/build/spec.py]: I'm happy about this now. The `Name` label, with '/' replaced by '-', looks like the correct name to use for the ImageStream.

Comment (mmilata) [inline osbs/build/spec.py]: Shouldn't the value of `trigger_imagestream_name` also have the slashes replaced? It looks like `get_imagestream_name_from_image` doesn't do such substitution.

File: inputs/simple_inner.json (hunk 2 above)
Comment (bkabrda) [inline inputs/simple_inner.json]: Hmm, not sure. I guess this really depends on how we define simple builds. I'm fine with only having triggers for prod builds, if others agree.

File: osbs/build/spec.py (hunk 3 above)
Comment (bkabrda) [inline osbs/build/spec.py]: @mmilata you're right, I'll fix that.

**Comment by bkabrda** [comment]
I added a commit that addresses issues found in previous commits. Everything should be fine now.

**Comment by bkabrda** [comment]
Good catch. I'll just do a force push to not introduce a new commit...

**Comment by twaugh** [comment]
LGTM

**Pull request merged**